### DISCUSSION TITLE
Frontend Validation

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#3376] [Research]
Validating inputs is a topic that has been raised multiple times in the past [^1], yet we still don't have a solution for this problem. I think it would make sense to settle on an approach.

### Why do we need a validation library

Rancher Desktop has multiple patterns for conveying invalid values in inputs. We can improve by applying a consistent design.

#### Examples

![image](https://user-images.githubusercontent.com/835961/201147989-0fafbce8-564e-49bf-b568-95a7ee829a51.png)

![image](https://user-images.githubusercontent.com/835961/201148347-e747cfe6-957c-465f-9dd4-1772fbb5eeba.png)

### Validation Libraries

Utilizing existing validation libraries can simplify the task of adding validation to existing forms, allowing us time to focus on the design and UX of validation messages. Some popular validation libraries for vue include 

* [Vuelidate](https://github.com/vuelidate/vuelidate)
* [VeeValidate](https://vee-validate.logaretm.com/v4/)

[^1]: https://github.com/rancher-sandbox/rancher-desktop/pull/2579#discussion_r926930482

**Comment by ericpromislow** [comment]
I would expect that the messages users get via the API or `rdctl` would be consistent with messages in the UI

**Reply by jandubois** [reply]
> I would expect that the messages users get via the API or `rdctl` would be consistent with messages in the UI

Does this means the validation needs to happen in the backend? Maybe this can simply piggyback on the `proposeSettings` API to validate all the settings as well?

**Comment by jandubois** [comment]
Issue #2540 has more details about various validations.

**Comment by gaktive** [comment]
@torchiaf to look at the validation libraries to see if they can help us out here.
